### ISSUE TITLE
`examples/stream` does not build 

### ISSUE THREAD

**Issue by dkg** [ggml-org/whisper.cpp#2626]
`examples/stream/README.md` says:

```
## Building

The `stream` tool depends on SDL2 library to capture audio from the microphone. You can build it like this:

```bash
# Install SDL2
# On Debian based linux distributions:
sudo apt-get install libsdl2-dev

# On Fedora Linux:
sudo dnf install SDL2 SDL2-devel

# Install SDL2 on Mac OS
brew install sdl2

make stream
```

Ensure you are at the root of the repo when running `make stream`. Not within the `examples/stream` dir
as the libraries needed like `common-sdl.h` are located within `examples`. Attempting to compile within
`examples/steam` means your compiler cannot find them and it gives an error it cannot find the file.

```bash
whisper.cpp/examples/stream$ make stream
g++     stream.cpp   -o stream
stream.cpp:6:10: fatal error: common/sdl.h: No such file or directory
    6 | #include "common/sdl.h"
      |          ^~~~~~~~~~~~~~
compilation terminated.
make: *** [<builtin>: stream] Error 1
```

I can confirm that running `make stream` fails in the way described above.  But running from the root of the repo yields:

```
0 dkg@bob:~/src/whisper/whisper.cpp$ make stream
make: *** No rule to make target 'stream'.  Stop.
2 dkg@bob:~/src/whisper/whisper.cpp$ 
```

**Comment by donivpx** [comment]
I had the same issue. Already addressed here and works for me - [Link](https://github.com/ggerganov/whisper.cpp/issues/2245#issuecomment-2176754333)

Try this from the root folder of Whisper
`cmake -DWHISPER_SDL2=ON .`

**Comment by Azkazar** [comment]
> I had the same issue. Already addressed here and works for me - [Link](https://github.com/ggerganov/whisper.cpp/issues/2245#issuecomment-2176754333)
> 
> Try this from the root folder of Whisper `cmake -DWHISPER_SDL2=ON .`

i dont understand the steps needed, i too have the same issues, im using windows and git bash, i too have installed SDL2

hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/whisper.cpp - Copy (master)
$ cmake -DWHISPER_SDL2=ON .
-- Building for: Visual Studio 17 2022
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19045.
-- The C compiler identification is MSVC 19.40.33808.0
-- The CXX compiler identification is MSVC 19.40.33808.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: C:/Program Files/Git/mingw64/bin/git.exe (found version "2.42.0.windows.2")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Found Threads: TRUE
-- ccache found, compilation results will be cached. Disable with GGML_CCACHE=OFF.
-- CMAKE_SYSTEM_PROCESSOR: AMD64
-- CMAKE_GENERATOR_PLATFORM:
-- Including CPU backend
-- Found OpenMP_C: -openmp (found version "2.0")
-- Found OpenMP_CXX: -openmp (found version "2.0")
-- Found OpenMP: TRUE (found version "2.0")
-- Performing Test HAS_AVX_1
-- Performing Test HAS_AVX_1 - Success
-- Performing Test HAS_AVX2_1
-- Performing Test HAS_AVX2_1 - Success
-- Performing Test HAS_FMA_1
-- Performing Test HAS_FMA_1 - Success
-- Performing Test HAS_AVX512_1
-- Performing Test HAS_AVX512_1 - Failed
-- Performing Test HAS_AVX512_2
-- Performing Test HAS_AVX512_2 - Failed
-- Adding CPU backend variant ggml-cpu: /arch:AVX2 GGML_AVX2;GGML_FMA;GGML_F16C
-- SDL2_INCLUDE_DIRS = C:/msys64/mingw64/include;C:/msys64/mingw64/include/SDL2
-- SDL2_LIBRARIES    = SDL2::SDL2main;SDL2::SDL2
-- Configuring done (18.0s)
-- Generating done (0.2s)
-- Build files have been written to: C:/Users/hp_5C/whisper.cpp - Copy

hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/whisper.cpp - Copy (master)
$ make stream
make: *** No rule to make target 'stream'.  Stop.

**Comment by dkg** [comment]
Thanks for the pointers, @donivpx.  What works for me on a debian system with `libsdl2-dev` installed is, from the repository root:

```
cmake -B build -DWHISPER_SDL2=ON
(cd build && make stream)
```

This produces the output binary in `build/bin/stream`

@Azkazar, Note that just running `cmake -DWHISPER_SDL2=ON .` will actually rewrite the top-level `Makefile`, which might not be what you want.  If you've already done that, then you might want to `git restore Makefile` (from the root of the git repo) to undo those changes.

**Comment by ggerganov** [comment]
I've updated the build instructions for the `examples/stream` tool:

https://github.com/ggerganov/whisper.cpp/tree/master/examples/stream

**Comment by Azkazar** [comment]
@ggerganov @dkg 

that doesnt work, bin/stream is not created

<details>
![Screenshot (2297)](https://github.com/user-attachments/assets/94dd306c-b35f-4a4b-91ff-e9999a75a43e)
</details>

using windows, cmake 3.31.2 (latest release 2024), git bash
<details>
<summary> Build details </summary>
<pre>
hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/Documents/agin/whisper.cpp (master)
$ cmake -B build -DWHISPER_SDL2=ON
-- Building for: Visual Studio 17 2022
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19045.
-- The C compiler identification is MSVC 19.40.33808.0
-- The CXX compiler identification is MSVC 19.40.33808.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022ommunity/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/20/Community/VC/Tools/MSVC/14.40.33807/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: C:/Program Files/Git/mingw64/bin/git.exe (found version "2.42.0.ndows.2")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - not found
-- Found Threads: TRUE
-- ccache found, compilation results will be cached. Disable with GGML_CCACHE=F.
-- CMAKE_SYSTEM_PROCESSOR: AMD64
-- CMAKE_GENERATOR_PLATFORM:
-- Including CPU backend
-- Found OpenMP_C: -openmp (found version "2.0")
-- Found OpenMP_CXX: -openmp (found version "2.0")
-- Found OpenMP: TRUE (found version "2.0")
-- Performing Test HAS_AVX_1
-- Performing Test HAS_AVX_1 - Success
-- Performing Test HAS_AVX2_1
-- Performing Test HAS_AVX2_1 - Success
-- Performing Test HAS_FMA_1
-- Performing Test HAS_FMA_1 - Success
-- Performing Test HAS_AVX512_1
-- Performing Test HAS_AVX512_1 - Failed
-- Performing Test HAS_AVX512_2
-- Performing Test HAS_AVX512_2 - Failed
-- Adding CPU backend variant ggml-cpu: /arch:AVX2 GGML_AVX2;GGML_FMA;GGML_F16
-- SDL2_INCLUDE_DIRS = C:/Users/hp_5C/Documents/installer program codingpentinSDL2-2.30.10/cmake/../include
-- SDL2_LIBRARIES    = SDL2::SDL2main;SDL2::SDL2
-- Configuring done (39.1s)
-- Generating done (0.6s)
-- Build files have been written to: C:/Users/hp_5C/Documents/agin/whisper.cppuild
</pre>
</details>

**Comment by ggerganov** [comment]
Hm, I'm not sure what is the problem. I don't have a Windows machine to test.

**Comment by Azkazar** [comment]
@ggerganov hello, due to the new update now i can build, but i cant run it even after changing the filename

<details>
<summary> Build & command log </summary>
<pre>

hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/Documents/agin/whisper.cpp (master)
$ cmake -B build -G "MinGW Makefiles" -DWHISPER_SDL2=ON
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The C compiler identification is GNU 14.2.0
-- The CXX compiler identification is GNU 14.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/msys64/ucrt64/bin/cc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/msys64/ucrt64/bin/c++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: C:/Program Files/Git/mingw64/bin/git.exe (found version "2.47.1.wi
ndows.1")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- ccache found, compilation results will be cached. Disable with GGML_CCACHE=OF
F.
-- CMAKE_SYSTEM_PROCESSOR: AMD64
-- Including CPU backend
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Adding CPU backend variant ggml-cpu: -march=native
-- SDL2_INCLUDE_DIRS = C:/msys64/mingw64/include;C:/msys64/mingw64/include/SDL2
-- SDL2_LIBRARIES    = SDL2::SDL2main;SDL2::SDL2
-- Configuring done (14.5s)
-- Generating done (0.9s)
-- Build files have been written to: C:/Users/hp_5C/Documents/agin/whisper.cpp/b
uild

hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/Documents/agin/whisper.cpp (master)
$ cd build && make stream
[ 50%] Building CXX object examples/deprecation-warning/CMakeFiles/stream.dir/de
precation-warning.cpp.obj
[100%] Linking CXX executable ..\..\bin\stream.exe
[100%] Built target stream

hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/Documents/agin/whisper.cpp/build (master)
$ ./build/bin/stream -m ./models/ggml-base.en.bin -t 6 --step 0 --length 30000 -vth 0.6
bash: ./build/bin/stream: No such file or directory

hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/Documents/agin/whisper.cpp/build (master)
$ cd ..

hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/Documents/agin/whisper.cpp (master)
$ ./build/bin/stream -m ./models/ggml-base.en.bin -t 6 --step 0 --length 30000 -vth 0.6

WARNING: The binary 'stream.exe' is deprecated.
 Please use 'whisper-stream.exe' instead.
 See https://github.com/ggerganov/whisper.cpp/tree/master/examples/deprecation-w
arning/README.md for more information.


hp_5C@DESKTOP-B2S5CM7 MINGW64 ~/Documents/agin/whisper.cpp (master)
$  ./build/bin/whisper-stream -m ./models/ggml-base.en.bin -t 6 --step 0 --length 30000 -vth 0.6

WARNING: The binary 'whisper-stream.exe' is deprecated.
 Please use 'whisper-whisper-stream.exe' instead.
 See https://github.com/ggerganov/whisper.cpp/tree/master/examples/deprecation-w
arning/README.md for more information.

</pre>
</details>
